float32 Unfiltered
float32 Filtered
int8 UsingFiltered
